File: leetcode/2225-find-players-with-zero-or-one-losses/findWinners.ts
function findWinners(matches: number[][]): number[][] {
  const lossesMap = new Map<number, number>();
  const result: number[][] = [[], []];

  for (const [winner, loser] of matches) {
    if (!lossesMap.has(winner)) {
      lossesMap.set(winner, 0);
    }

    if (!lossesMap.has(loser)) {
      lossesMap.set(loser, 1);
    } else {
      lossesMap.set(loser, lossesMap.get(loser) + 1);
    }
  }

  for (const [key, count] of lossesMap) {
    if (count >= 2) {
      continue;
    }

    if (count === 0) {
      result[0].push(key);
    } else {
      result[1].push(key);
    }
  }

  result[0].sort((a, b) => a - b);
  result[1].sort((a, b) => a - b);

  return result;
}

export default findWinners;

File: leetcode/2225-find-players-with-zero-or-one-losses/findWinners.go
package leetcode

import "slices"

func findWinners(matches [][]int) [][]int {
	var lossesMap = map[int]int{}
	var result = [][]int{{}, {}}

	for _, match := range matches {
		var winner, loser = match[0], match[1]

		if _, ok := lossesMap[winner]; !ok {
			lossesMap[winner] = 0
		}

		lossesMap[loser]++
	}

	for key, count := range lossesMap {
		if count >= 2 {
			continue
		}

		result[count] = append(result[count], key)
	}

	slices.Sort(result[0])
	slices.Sort(result[1])

	return result
}
